subprojects/snapshots/src/main/java/org/gradle/internal/snapshot/impl/DirectorySnapshotter.java

```diff
@@ -235,7 +298,10 @@ public FileVisitResult postVisitDirectory(Path dir, @Nullable IOException exc) {
             if (isNotFileSystemLoopException(exc)) {
                 throw new UncheckedIOException(String.format("Could not read directory path '%s'.", dir), exc);
             }
-            builder.postVisitDirectory(AccessType.DIRECT);
+            AccessType accessType = (!symbolicLinkMappings.isEmpty() && symbolicLinkMappings.getFirst().target.equals(dir.toString()))
+                ? AccessType.VIA_SYMLINK
+                : AccessType.DIRECT;
```

Comment:
What do you think about adding something like `AccessType.viaSymlink(boolean)`?

Reply:
Added.